realm/src/androidTest/java/io/realm/internal/JNITransactions.java

```diff
@@ -376,4 +378,13 @@ public void testAddEmptyRowWithPrimaryKeyLong() {
         assertEquals(1, t.size());
         assertEquals(42, t.getRow(rowIndex).getLong(0));
     }
+
+    public void testPrimaryKeyTableMigration() throws IOException {
```

Comment:
It is sufficient to test with two different old primary key layouts: 0 primary keys and 1 primary key (triggering the loop).